packages/builder-vite/mdx-plugin.ts

```diff
@@ -9,7 +18,50 @@ import { createCompiler } from '@storybook/csf-tools/mdx';
  *
  * @see https://github.com/storybookjs/storybook/blob/next/addons/docs/docs/recipes.md#csf-stories-with-arbitrary-mdx
  */
-export function mdxPlugin() {
+export async function mdxPlugin(config: any): Promise<Plugin | Plugin[]> {
+  const { features } = config;
+
+  if (features?.previewMdx2) {
+    logger.info('using MDX2');
+
+    // @ts-expect-error the types of mdx2-csf don't seem quite right
+    const { compile } = await import('@storybook/mdx2-csf');
+    let reactRefreshPlugin: Plugin | undefined;
+
+    return {
+      name: 'storybook-builder-vite:mdx2-plugin',
+      configResolved({ plugins }) {
+        // @vitejs/plugin-react-refresh has been upgraded to @vitejs/plugin-react,
+        // and the name of the plugin performing `transform` has been changed from 'react-refresh' to 'vite:react-babel',
+        // to be compatible, we need to look for both plugin name.
+        // We should also look for the other plugins names exported from @vitejs/plugin-react in case there are some internal refactors.
+        const reactRefreshPlugins = plugins.filter(
+          (p) =>
+            p.name === 'react-refresh' ||
+            p.name === 'vite:react-babel' ||
+            p.name === 'vite:react-refresh' ||
+            p.name === 'vite:react-jsx'
+        );
+        reactRefreshPlugin = reactRefreshPlugins.find((p: any) => p.transform);
+      },
+      async transform(src: string, id: string, ssr) {
+        if (!/\.mdx?$/.test(id)) return src;
+        const code = await compile(src, { skipCsf: true });
```

Comment:
We won't want to `skipCsf` in the final version, but it avoids some errors for now.